### ISSUE TITLE
How can I add metrics in losses.py?

### ISSUE THREAD

**Issue by Wusir2018** [nilmtk/nilmtk#903]
hello everyone,
I try to add metric `recall score` in `losses.py`, but it doesn't work!
Could you help me?
Should custom metric be added to `losses.py`?

I change the `losses.py`:
```
from sklearn.metrics import mean_squared_error, mean_absolute_error, f1_score, r2_score,recall_score,precision_score
import numpy as np

def mae(app_gt,app_pred):
    return mean_absolute_error(app_gt,app_pred)

def rmse(app_gt, app_pred):
    return mean_squared_error(app_gt,app_pred)**(.5)

def f1score(app_gt, app_pred):
    threshold = 10
    gt_temp = np.array(app_gt)
    gt_temp = np.where(gt_temp<threshold,0,1)
    pred_temp = np.array(app_pred)
    pred_temp = np.where(pred_temp<threshold,0,1)

    return f1_score(gt_temp, pred_temp)

def recallscore(app_gt, app_pred):
    threshold = 10
    gt_temp = np.array(app_gt)
    gt_temp = np.where(gt_temp<threshold,0,1)
    pred_temp = np.array(app_pred)
    pred_temp = np.where(pred_temp<threshold,0,1)

    return recall_score(gt_temp, pred_temp)

```

error:
```
FHMM partial_fit end.................
Finished training for  FHMM
Joint Testing for all algorithms
Loading data for  Datport  dataset
Loading data for meter ElecMeterID(instance=2, building=1, dataset='REDD')     
Done loading data all meters for this chunk.
Dropping missing values
Generating predictions for : FHMM
**Loss function  recallscore  is not supported currently!**


```

**Comment by eddie-LA** [comment]
@PMeira I am also interested in this. I have implemented an 'estimated accuracy' from this paper "Nonintrusive Load Monitoring (NILM) Performance Evaluation, A unified approach for accuracy reporting by Stephen Makonin and Fred Popowich" in metrics.py but I quickly found out I need to use losses.py in the API. 
If this is answered, I can submit some pull requests to have that included along with some other patches and useful things I have been working on.

**Comment by levaphenyl** [comment]
If you are using the `API` class, you can directly define the function in your main Python file and refer to its name in the experiment parameters.
What [this line](https://github.com/nilmtk/nilmtk/blob/master/nilmtk/api.py#L350) does is a search in the entire namespace for a function of that name, e.g. `"f1score"` will be found in `nilmtk.losses`.

Two solutions then:
1. You open a PR updating `losses.py` with your state-of-the-art metric because you love NILMTK and want to improve it.
2. You are testing a metric and just define it in the same script that runs the experiment.
